### DISCUSSION TITLE
Managing queue locally result an unstable behavior.

### DISCUSSION THREAD

**Discussion by KobiAms** [gilbarbara/react-spotify-web-playback#158] [General]
Hi Gil.

I managing the songs queue locally, so I simply replace the uris object when song is ended.
I notice that the api/.../play endpoint not too much stable and sometimes returns a 502 Bad gateway http error.
I'm guessing its due to load on Spotify servers.
It is possible please to add an onError prop to the player that invoked when an error occurred so I can invoked the request again? or maybe a "retry" policy?
Any other suggestion that deals with api errors will be great.

BTW, the last update which exposed the player and api is really great! thank you. 

<img width="458" alt="Screenshot 2023-05-04 at 22 42 08" src="https://user-images.githubusercontent.com/59447829/236312717-7c83b188-12d1-478b-b025-6d18c9d089cd.png">

found that in the wild: https://community.spotify.com/t5/Spotify-for-Developers/Web-API-502-error-on-me-player-play-for-a-short-period-after/m-p/5400268

**Comment by gilbarbara** [comment]
Hey @KobiAms 

You don't need to replace the `uris` for every track if you have several songs to play in sequence.
Set an array of multiple uris and let the player handle the queue instead.

About a new prop to handle Spotify API errors, I'm not planning to add something like this right now.
In my experience, when the API returns 502, retrying right away won't change the result.
